Three breaking changes (#36787)
dotnet/docs: docs/core/compatibility/extensions/8.0/directoryservices-package.md

---
title: "Breaking change: DirectoryServices package no longer references System.Security.Permissions"
description: Learn about the .NET 8 breaking change in .NET extensions where the System.DirectoryServices package no longer references the System.Security.Permissions package.
ms.date: 08/22/2023
---
# DirectoryServices package no longer references System.Security.Permissions

The `System.DirectoryServices` package no longer references the `System.Security.Permissions` package.

## Version introduced

.NET 8 Preview 3

## Previous behavior

The `System.DirectoryServices` package referenced the `System.Security.Permissions` package.

## New behavior

> Starting in .NET 8, the `System.DirectoryServices` package *does not reference* the `System.Security.Permissions` package.

Reviewer: ```suggestion
Starting in .NET 8, the `System.DirectoryServices` package *doesn't reference* the `System.Security.Permissions` package.
```

Author: ```suggestion
Starting in .NET 8, the `System.DirectoryServices` package *does not* reference the `System.Security.Permissions` package.
```

## Type of breaking change

This change can affect [source compatibility](../../categories.md#source-compatibility).

## Reason for change

This change avoids a dependency on `System.Drawing.Common` when `System.DirectoryServices` is referenced, which is primarily an issue for non-Windows operating systems.

The dependency on `System.Drawing.Common` was caused by the following package dependencies:

```txt

> System.DirectoryServices ->
> System.Security.Permissions ->
> System.Windows.Extensions ->
> System.Drawing.Common

Reviewer: ```suggestion
System.DirectoryServices
 └──System.Security.Permissions
      └──System.Windows.Extensions
           └──System.Drawing.Common
```
